### PR TITLE
Enhancements to cqlsh copy tests for CASSANDRA-9304

### PR DESCRIPTION
**Pull request by stef1927**

### CODE REVIEW

File: cqlsh_tests/cqlsh_copy_tests.py [hunk 1]
```diff
@@ -16,12 +16,18 @@
...
 from tools import rows_to_list, since
 
 DEFAULT_FLOAT_PRECISION = 5  # magic number copied from cqlsh script
 DEFAULT_TIME_FORMAT = '%Y-%m-%d %H:%M:%S'  # based on cqlsh script; timezone stripped
 
+PARTITIONERS = dict()
+PARTITIONERS["murmur3"] = "org.apache.cassandra.dht.Murmur3Partitioner"
+PARTITIONERS["random"] = "org.apache.cassandra.dht.RandomPartitioner"
+PARTITIONERS["byte"] = "org.apache.cassandra.dht.ByteOrderedPartitioner"
+PARTITIONERS["order"] = "org.apache.cassandra.dht.OrderPreservingPartitioner"
```
Comment (mambocab) [inline cqlsh_tests/cqlsh_copy_tests.py]: Using a dictionary literal is better style here:

```python
PARTITIONERS = {
    "murmur3": "org.apache.cassandra.dht.Murmur3Partitioner",
    "random": "org.apache.cassandra.dht.RandomPartitioner",
    "byte": "org.apache.cassandra.dht.ByteOrderedPartitioner",
    "order": "org.apache.cassandra.dht.OrderPreservingPartitioner"
}
```

File: cqlsh_tests/cqlsh_copy_tests.py [hunk 2]
```diff
@@ -30,6 +36,11 @@ class CqlshCopyTest(Tester):
     Tests the COPY TO and COPY FROM features in cqlsh.
     @jira_ticket CASSANDRA-3906
     """
+
+    def __init__(self, *args, **kwargs):
+        Tester.__init__(self, *args, **kwargs)
+        self.tempfile = None
```
Comment (mambocab) [inline cqlsh_tests/cqlsh_copy_tests.py]: I don't see a need to initialize `self.tempfile` here. If you're running into problems in `tearDown`, then catch an `AttributeError` when you dereference it.

File: cqlsh_tests/cqlsh_copy_tests.py [hunk 3]
```diff
@@ -46,10 +57,16 @@ def tearDown(self):
 
         super(CqlshCopyTest, self).tearDown()
 
-    def prepare(self):
+    def prepare(self, nodes=1, partitioner="murmur3"):
         if not self.cluster.nodelist():
-            self.cluster.populate(1).start(wait_for_binary_proto=True)
-        self.node1, = self.cluster.nodelist()
+            p = PARTITIONERS.get(partitioner)
```
Comment (mambocab) [inline cqlsh_tests/cqlsh_copy_tests.py]: If `PARTITIONERS` doesn't have key `partitioner`, `p` will be `None` here. I don't think that'll play well with `set_partitioner`. Is that intentional? You could fail earlier in that case by using `PARTITIONERS[partitioner]`.

**Comment by mambocab** [comment]
Other than my little nits, these look great, thank you!

**Comment by stef1927** [comment]
Thank you for your comments, they should be fixed now!

**Comment by mambocab** [comment]
Excellent, thank you! Merging.

**Pull request merged**